mobile/src/main/java/org/openhab/habdroid/ui/ViewExtensions.kt

```diff
@@ -111,3 +122,26 @@ fun RemoteViews.duplicate(): RemoteViews {
         clone()
     }
 }
+
+fun MaterialButton.setTextAndIcon(connection: Connection, mapping: LabeledValue) {
+    val iconUrl = mapping.icon?.toUrl(context, true)
+    if (iconUrl == null) {
+        icon = null
+        text = mapping.label
+        return
+    }
+    CoroutineScope(Dispatchers.IO + Job()).launch {
+        val drawable = try {
+            connection.httpClient.get(iconUrl, caching = HttpClient.CachingMode.DEFAULT)
+                .asBitmap(100, 0, ImageConversionPolicy.ForceTargetSize).response
```

Comment:
Probably should use a `dimen` resource for the bitmap size (16dp?)

Reply:
Sure. The `100` was just added for faster development and I forgot to replace it...